src/07-theory_operations.adoc

```diff
@@ -13,60 +13,30 @@ devices.
-2. The host supervisor domain manager (e.g. the host VMM) enumerates all
-   platform IOMMUs as PCIe devices.
-3. The host IOMMU driver is loaded and probed.
-4. The host IOMMU driver allocates MSI vectors for the trusted domain
-   IOMMUs. Those vectors must point to the untrusted IOMMU driver as the TSM can
-   not handle external interrupts.
-5. The host supervisor domain manager registers the IOMMUs with the TSM by
-   calling the `sbi_covh_register_iommu()` `COVH` function. The TSM gets the
-   allocated MSI vectors and configures the trusted IOMMU `s_msi_cfg_table`
-   register accordingly.
-6. When a trusted IOMMU sends an MSI, the untrusted IOMMU driver handles it and
-   notifies the TSM about a pending trusted IOMMU MSI by calling the
-   `sbi_covh_notify_iommu_msi()` `COVH` function.
-7. The TSM verifies that there is a pending MSI by reading the IOMMU `s_ipsr`
-   register, and handles the interrupts as needed.
-
-The host IOMMU driver may be malicious and attempt to trick the TSM by either
-invoking the `sbi_covh_notify_iommu_msi()` `COVH` function while there are no
-pending MSI from the trusted IOMMU or not invoking it when there actually is one
-such pending interrupt. In the former case, the TSM can verify that there really
-is a pending MSI by checking the trusted IOMMU status registers. The latter case
-could cause a denial of service, which is not in scope of the confidential
-computing threat model. If the TSM must ensure that a given command is completed
-without having to rely on untrusted IOMMU driver MSI notifications, it can queue
-an `IOFENCE.C` command after the desired command and check for the `cqh`
-advancing past the `IOFENCE.C` command index.
+=== Platform Management
+
+==== IOMMU Assignment
+
+For CoVE-IO, a TSM relies on being granted exclusive control over one IOMMU
+instance on the platform. An IOMMU instance refers to any instantiation of a
+RISC-V IOMMU Register Programming Interface (RPI), which could range from an
+actual IOMMU instance to an MTT-enforced, memory-mapped access to a physically
+partitioned IOMMU.
+
+This exclusvely assigned IOMMU RPI allows the TSM to enforce the integrity of
+address translations and protection from DMA into confidential memory, as well
+as interrupts originating from TDIs assigned to TVMs under the TSM control.
+
+When creating a Supervisor Domain (SD), the RDSM can optionally establish
+exclusive access to an IOMMU RPI for the SD TSM, prior to loading and scheduling
+said TSM. The RDSM creates <<Smmpt>> mappings to assign the IOMMU RPI to the SD
+and isolate it from any other domain, including the hosting SD.
+
+When the hosting SD requests a TDI to be bound to a TVM, the RDSM first programs
+the I/O-MPT Supervisor Domain Classifier (SDCL) to map DMA requests originating
+from the TDI to the TVM's SD and its IOMMU. Then the TSM can program the IOMMU
+RPI with the desired IOVA translations and interrupt remappings for the assigned
+TDI.
```

Comment:
these two things don't have to occur in that order, do they? as long as they both are performed by the time the TDI is being readied for DMA - it is fine to be done in any order.

Reply:
That's right. I changed that section a little bit to highlight the fact that both things could happen in parallel and also that the IOMMU mappings must not be enabled until the TVM accepts the TDI.